src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceService.ChangeNamespaceCodeAction.cs

```diff
@@ -0,0 +1,526 @@
+                    if (refLoc.Alias != null)
+                    {
+                        continue;
+                    }
+
+                    // Other documents in the solution might have changed after we calculated those ReferenceLocation, 
+                    // so we can't trust anything to be still up-to-date except their spans.
+
+                    // Get inner most node in case of type used as a base type. e.g.
+                    //
+                    //      public class Foo {}
+                    //      public class Bar : Foo {}
+                    //
+                    // For the reference to Foo where it is used as a base class, the BaseTypeSyntax and the TypeSyntax
+                    // have exact same span.
+
+                    var refNode = root.FindNode(refLoc.Location.SourceSpan, getInnermostNodeForTie: true);
+                    if (syncNamespaceService.TryGetReplacementReferenceSyntax(
+                        refNode, namespaceParts, syntaxFacts, out var oldNode, out var newNode))
+                    {
+                        editor.ReplaceNode(oldNode, newNode.WithAdditionalAnnotations(Simplifier.Annotation));
+                    }
+
+                    // Use a dummy import node to figure out which container the new import will be added to.
+                    var container = addImportService.GetImportContainer(root, refNode, dummyImport);
+                    containers.Add(container);
+                }
+
+                foreach(var container in containers)
+                {
+                    editor.TrackNode(container);
+                }
+
+                document = editor.GetChangedDocument();
+                root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                return (document, containers.Select(c => root.GetCurrentNode(c)).ToImmutableArray());
+            }
+
+            private async Task<Solution> RemoveUnnecessaryImportsAsync(
+                Solution solution, 
+                ImmutableArray<DocumentId> ids, 
+                ImmutableArray<string> names, 
+                CancellationToken cancellationToken)
+            {
+                var LinkedDocumentsToSkip = PooledHashSet<DocumentId>.GetInstance();
+                foreach (var id in ids)
+                {
+                    if (LinkedDocumentsToSkip.Contains(id))
+                    {
+                        continue;
+                    }
+
+                    var document = solution.GetDocument(id);
+                    LinkedDocumentsToSkip.AddRange(document.GetLinkedDocumentIds());
+
+                    document = await RemoveUnnecessaryImportsAsync(
+                        document, 
+                        CreateImports(document, names, withFormatterAnnotation: false), 
+                        cancellationToken)
+                        .ConfigureAwait(false);
```

Comment:
no need to wrap .ConfigureAwait.  And it looks super weird to have this dangling .ConfigureAwait there.